pkg/kclient/pods.go

```diff
@@ -0,0 +1,118 @@
+package kclient
+
+import (
+	"io"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/openshift/odo/pkg/log"
+	"github.com/pkg/errors"
+
+	// api resource types
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+const (
+	// waitForPodTimeOut controls how long we should wait for a pod before giving up
+	waitForPodTimeOut = 240 * time.Second
+)
+
+// WaitAndGetPod block and waits until pod matching selector is in in Running state
+// desiredPhase cannot be PodFailed or PodUnknown
+func (c *Client) WaitAndGetPod(watchOptions metav1.ListOptions, desiredPhase corev1.PodPhase, waitMessage string) (*corev1.Pod, error) {
```

Comment:
Should we accept multiple desired phases? eg. Running or Succeeded?

Reply:
Since the corresponding function in `occlient` only accepts one desiredPhase, I'd prefer to leave it as-is for now.